context/app/static/js/hooks/useSearchData.js

```diff
@@ -0,0 +1,35 @@
+import { useState, useEffect } from 'react';
+import { getAuthHeader } from 'js/helpers/functions';
+
+function useSearchData(query, elasticsearchEndpoint, nexusToken) {
+  const [searchData, setSearchData] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
+
+  useEffect(() => {
+    async function getAndSetEntity() {
+      const authHeader = getAuthHeader(nexusToken);
+      const response = await fetch(elasticsearchEndpoint, {
+        method: 'POST',
+        body: query,
+        headers: {
+          'Content-Type': 'application/json',
+          ...authHeader,
+        },
+      });
+      if (!response.ok) {
+        console.error('Search API failed', response);
+        return;
+      }
+      const results = await response.json();
+      // eslint-disable-next-line no-underscore-dangle
+      const resultEntity = results.hits.hits;
+      setSearchData(resultEntity);
```

Comment:
```suggestion
      setSearchHits(results.hits.hits);
```
`data` is really vague, and I think the search hits will be multiple "entities", not just one?